File: Practica-4/Practica4.java
package practica4;

/**
 *
 * @author DAVID
 */
public class Practica4 {

    double calculararea (double radio){
        return Math.PI * radio * radio;
    } 
       
    double calculararea (double base, double altura){
        return base * altura;
    }
    
    double calculararea(double base, double altura, double triangulo){
        return (base * altura) / 2;
    }

    double calculararea(double base1, double base2, double altura, boolean trapecio){
        return ((base1 + base2) * altura) / 2;
    }

    double calculararea(double lado, double apotema, boolean pentagono){
        return (5 * lado * apotema) / 2;
    }

    public static void main(String[] args) {
        Practica4 f1 = new Practica4();
        Practica4 f2 = new Practica4();   
        Practica4 f3 = new Practica4();
        Practica4 f4 = new Practica4();
        Practica4 f5 = new Practica4();
        
        System.out.println("Circulo: "+ f1.calculararea(1));
        System.out.println("Rectangulo: "+ f2.calculararea(2,3));
        System.out.println("Triangulo Rectangulo: "+ f3.calculararea(3,5));
        System.out.println("Trapecio: "+ f4.calculararea(2,3,4));
        System.out.println("Pentagono: "+ f5.calculararea(2,4));
     }   
}

File: Practica-4/Practica4.py
import math

class Practica4:

    def calcular_area(self, radio=None, base=None, altura=None, tipo=None):
        if radio is not None:
            return math.pi * radio * radio
        elif base is not None and altura is not None and tipo is None:
            return base * altura
        elif base is not None and altura is not None and tipo == "triangulo":
            return (base * altura) / 2
        elif base is not None and altura is not None:
            return ((base + altura) * altura) / 2
        elif base is not None and altura is not None:
            return (5 * base * altura) / 2
        else:
            return 0

if __name__ == "__main__":
    f = Practica4()

    print("Círculo: ", f.calcular_area(radio=1))
    print("Rectángulo: ", f.calcular_area(base=2, altura=3))
    print("Triángulo Rectángulo: ", f.calcular_area(base=3, altura=5, tipo="triangulo"))
    print("Trapecio: ", f.calcular_area(base=2, altura=3))
    print("Pentágono: ", f.calcular_area(base=2, altura=4))
